### PR TITLE
update guice from 4.0 to 4.2.2

### PR DESCRIPTION
**Pull request by based2**
https://mvnrepository.com/artifact/com.google.inject/guice

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
retriggering CI

**Comment by MRamonLeon** [comment]
I see in the log of the tests, maybe it's guice-related and not infra problems:
 
```
at java.base/java.lang.Thread.run(Thread.java:834)
1.646 [id=23]	WARNING	h.ExtensionFinder$GuiceFinder$SezpozModule#configure: Failed to load hudson.ExtensionListTest
java.lang.NoSuchMethodError: 'void com.google.common.base.Preconditions.checkArgument(boolean, java.lang.String, java.lang.Object)'
at com.google.inject.Key.ensureRetainedAtRuntime(Key.java:341)
at com.google.inject.Key.strategyFor(Key.java:317)
at com.google.inject.Key.get(Key.java:224)
at hudson.ExtensionFinder$GuiceFinder$SezpozModule.configure(ExtensionFinder.java:543)
at com.google.inject.AbstractModule.configure(AbstractModule.java:61)
at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:344)
at com.google.inject.spi.Elements.getElements(Elements.java:103)
at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:137)
at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:103)
at com.google.inject.Guice.createInjector(Guice.java:87)
at com.google.inject.Guice.createInjector(Guice.java:69)
at com.google.inject.Guice.createInjector(Guice.java:59)
at hudson.ExtensionFinder$GuiceFinder.<init>(ExtensionFinder.java:290)
```

**Comment by fcojfernandez** [comment]
There are 1063 failing tests and one of the last log messages says there was an Out Of Memory Error. Could you have a look a this, @based2 ?

```
[2019-08-19T06:32:44.174Z] [ERROR] Tests run: 1090, Failures: 0, Errors: 1063, Skipped: 24
[2019-08-19T06:32:44.174Z] [INFO] 
[2019-08-19T06:32:46.788Z] [INFO] ------------------------------------------------------------------------
[2019-08-19T06:32:46.788Z] [INFO] Reactor Summary for Jenkins main module 2.191-rc28602.f20028437190:
[2019-08-19T06:32:46.788Z] [INFO] 
[2019-08-19T06:32:46.788Z] [INFO] Jenkins main module ................................ SUCCESS [  9.952 s]
[2019-08-19T06:32:46.788Z] [INFO] Jenkins cli ........................................ SUCCESS [ 24.138 s]
[2019-08-19T06:32:46.788Z] [INFO] Jenkins core ....................................... SUCCESS [02:36 min]
[2019-08-19T06:32:46.788Z] [INFO] Jenkins war ........................................ SUCCESS [ 43.437 s]
[2019-08-19T06:32:46.788Z] [INFO] Jenkins core tests POM ............................. SUCCESS [ 12.593 s]
[2019-08-19T06:32:46.788Z] [INFO] Tests for Jenkins core ............................. FAILURE [36:32 min]
[2019-08-19T06:32:46.788Z] [INFO] JDK8-only tests for Jenkins core ................... SKIPPED
[2019-08-19T06:32:46.788Z] [INFO] ------------------------------------------------------------------------
[2019-08-19T06:32:46.788Z] [INFO] BUILD FAILURE
[2019-08-19T06:32:46.788Z] [INFO] ------------------------------------------------------------------------
[2019-08-19T06:32:46.788Z] [INFO] Total time:  40:55 min
[2019-08-19T06:32:46.788Z] [INFO] Finished at: 2019-08-19T06:32:46Z
[2019-08-19T06:32:46.788Z] [INFO] ------------------------------------------------------------------------
[2019-08-19T06:32:46.788Z] [ERROR] Java heap space -> [Help 1]
[2019-08-19T06:32:46.788Z] java.lang.OutOfMemoryError: Java heap space
```

**Comment by based2** [comment]
Seems that guice and guava must move in sync.
https://stackoverflow.com/questions/42206440/java-lang-nosuchmethoderror-com-google-common-base-preconditions-checkargument

"
I had the same issue too and I have solved it by updating the version of com.google.inject.guice from 4.1.0 to 4.2.2. I have realized that, guice was using guava with version 19.0 and it caused the bug for no such method for checkArgument. By updating the version of guice, the problem solved for me. You can check if there is another library using guava by "mvn dependency:tree" and you can update it. – irukeru Aug 6 at 8:21"

File: pom.xml [hunk 1]
```diff
@@ -90,8 +90,8 @@ THE SOFTWARE.
     <project.patchManagement.url>https://api.github.com</project.patchManagement.url>
     <patch.tracker.serverId>jenkins-jira</patch.tracker.serverId>
 
-    <guavaVersion>11.0.1</guavaVersion>
-    <slf4jVersion>1.7.25</slf4jVersion>
+    <guavaVersion>>25.1-jre</guavaVersion>
```
Comment (fcojfernandez) [inline pom.xml]: Extra `>`
```suggestion
    <guavaVersion>25.1-jre</guavaVersion>
```

**Review by fcojfernandez — CHANGES_REQUESTED** [review]
pom.xml has a conflict now

**Comment by oleg-nenashev** [comment]
I propose closing it if there is no interest to finalize it.
I am generally against unjustified dependency bumps while we do not have full integration testing setup in the core

**Comment by varyvol** [comment]
Given no answer since August and it was already proposed for close two weeks ago, we are closing this Pull Request.

Thanks a lot for your understanding and please reopen in case you consider to continue working on it.

**Pull request closed without merging**